CONTRIBUTING.md

```diff
@@ -1,33 +1,61 @@
 
-## How to report a bug
+From a release planning standpoint, we use github milestones that match to tags in the Lyra repository. Each milestone aims for a usable (if not necessarily *complete*) chunk of user value, plus bug fixes and other improvements that come in during the course of its development. The [Milestones page](https://github.com/lyraproj/lyra/milestones) for the project shows the upcoming targets and their themes; contributions that work towards that theme are extremely valuable!
 
-Open a "[Bug report](https://github.com/lyraproj/lyra/issues/new?template=bug_report.md)" issue in this repo.
+## Filing issues
+
+If you're trying to use Lyra and have run into a problem, first try [searching the existing issues](https://github.com/lyraproj/lyra/issues?q=) to see if someone has already reported a similar problem. If not, you can use the "[bug report template](https://github.com/lyraproj/lyra/issues/new?template=bug_report.md)" to open an issue.
+
+If you've got an idea for an enhancement or new feature, first take a look at the [ROADMAP Project board](https://github.com/lyraproj/lyra/projects/1) where we keep a backlog of potential enhancements. If there's something similar to what you're thinking about, it's super helpful to add on to the existing issue with your ideas and use-cases. If you don't see anyting similar, open a "[Feature request](https://github.com/lyraproj/lyra/issues/new?template=feature_request.md)" issue in this repo.
+
+## Submitting Changes
 
-## How to suggest a new feature
 
-Open a "[Feature request](https://github.com/lyraproj/lyra/issues/new?template=feature_request.md)" issue in this repo.
+If you're thinking about making substantial code changes, please file an issue describing the problem and your proposed solution before you start working on it. If you mention @lyraproj/committers in your issue, it'll notify the core team to ensure someone engages with the proposal.
+
+Once you're ready to submit your change, follow the usual github PR process. We (try to!) follow the [codelikeagirl guidelines](https://code.likeagirl.io/useful-tips-for-writing-better-git-commit-messages-808770609503) for writing good commit messages: format for short lines, use the imperative mood ("Add X to Y"), describe before and after state in the commit message body.
+
+<!-- We currently require a [Developer Certificate of Origin](https://developercertificate.org/) signoff for commits. See [this explanation](https://helm.sh/blog/helm-dco/) from the Helm project to understand the rationale behind the DCO. Actually adding the signoff is as simple as adding the `-s` flag to your commits, or configuring it in your git-config:
+
+```
+git commit -s -m "Add feature Foo to Lyra"
+# or permanently
+git config --add format.signoff true
```

Comment:
I don't think this does what you want it to.

Reply:
If we also accepted GPG signed commits, the equivalent is
```
[commit]
     gpgsign = true
```

Reply:
No `format.signoff` is for patch only.

Reply:
There *is* a `prepare-commit-msg` hook that you can install, but that's a pita. https://stackoverflow.com/a/46536244/10958549

Reply:
thanks i'll remove this.